server/datastore/mysql/hosts_test.go

```diff
@@ -7530,3 +7530,72 @@ func testListHostsWithPagination(t *testing.T, ds *Datastore) {
 	require.NoError(t, err)
 	require.Equal(t, hostCount, count)
 }
+
+func testLastRestarted(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+
+	// Arbitrary value
+	const uptimeVal = 16691000000000
+	now := time.Now()
+	newHostFunc := func(name string, uptimeZero bool) (*fleet.Host, time.Time) {
+		newHost := &fleet.Host{
+			DetailUpdatedAt: now,
+			LabelUpdatedAt:  now,
+			PolicyUpdatedAt: now,
+			SeenTime:        now,
+			NodeKey:         ptr.String(name),
+			UUID:            name,
+			Hostname:        "foo.local." + name,
+		}
+
+		var expectedLastRestartedAt time.Time
+
+		if uptimeZero {
+			newHost.Uptime = 0
+		} else {
+			newHost.Uptime = uptimeVal
+			// Rounding to nearest second because the SQL query does integer division.
+			expectedLastRestartedAt = newHost.DetailUpdatedAt.Add(time.Duration(-newHost.Uptime)).Round(time.Second).UTC()
+		}
+
+		host, err := ds.NewHost(ctx, newHost)
+		require.NoError(t, err)
+		require.NotNil(t, host)
+		return host, expectedLastRestartedAt
+	}
+
+	hostCount := 10
```

Comment:
Any reason you create 10 instead of 2?

Reply:
Not particularly, seemed like a nice round number lol